backend/src/middleware/validation/index.js

```diff
@@ -0,0 +1,31 @@
+import { UserInputError } from 'apollo-server'
+
+const USERNAME_MIN_LENGTH = 3
+
+const validateUsername = async (resolve, root, args, context, info) => {
+  if (args.name && args.name.length >= USERNAME_MIN_LENGTH) {
+    /* eslint-disable-next-line no-return-await */
+    return await resolve(root, args, context, info)
```

Comment:
are we updating the user twice here? once in the validation middleware and once on line 15 of the `userMiddleware`?

Reply:
No, calling `resolve` will continue the chain of middlewares until you hit the final resolver.